src/main/java/org/sagebionetworks/bridge/models/schedules2/adherence/participantschedule/ParticipantScheduleGenerator.java

```diff
@@ -0,0 +1,128 @@
+import com.google.common.collect.Lists;
+import com.google.common.collect.Multimap;
+
+public class ParticipantScheduleGenerator {
+    private static final LocalDate EARLIEST_LOCAL_DATE = LocalDate.parse("1900-01-01");
+    private static final LocalDate LATEST_LOCAL_DATE = LocalDate.parse("9999-12-31");
+
+    public static final ParticipantScheduleGenerator INSTANCE = new ParticipantScheduleGenerator();
+
+    public ParticipantSchedule generate(AdherenceState state, Timeline timeline) {
+        Multimap<LocalDate, ScheduledSession> chronology = LinkedHashMultimap.create();
+        LocalDate earliestDate = LATEST_LOCAL_DATE;
+        LocalDate latestDate = EARLIEST_LOCAL_DATE;
+        
+        for (ScheduledSession schSession : timeline.getSchedule()) {
+            String eventId = schSession.getStartEventId();
+            DateTime eventTimestamp = state.getEventTimestampById(eventId);
+            if (eventTimestamp == null) {
+                continue;
+            }
+            Integer days = state.getDaysSinceEventById(eventId);
+            LocalDate startDate = eventTimestamp.plusDays(schSession.getStartDay()).toLocalDate();
+            LocalDate endDate = eventTimestamp.plusDays(schSession.getEndDay()).toLocalDate();
+            AdherenceRecord record = state.getAdherenceRecordByGuid(schSession.getInstanceGuid());
+            SessionCompletionState schState = AdherenceUtils.calculateSessionState(
+                    record, schSession.getStartDay(), schSession.getEndDay(), days);
+            
+            ScheduledSession.Builder builder = schSession.toBuilder();
+            builder.withStartDate(startDate);
+            builder.withEndDate(endDate);
+            // We do not show state for persistent sessions, because it makes no sense (it's the most recent
+            // timestamp, but that's just confusing).
+            if (!TRUE.equals(schSession.isPersistent())) {
+                builder.withState(schState);
+            }
+            if (startDate.isBefore(earliestDate)) {
+                earliestDate = startDate;
+            }
+            if (endDate.isAfter(latestDate)) {
+                latestDate = endDate;
+            }
+            for (ScheduledAssessment schAssessment : schSession.getAssessments()) {
+                AdherenceRecord asmtRecord = state.getAdherenceRecordByGuid(schAssessment.getInstanceGuid());    
+                SessionCompletionState asmtState = AdherenceUtils.calculateSessionState(
+                        asmtRecord, schSession.getStartDay(), schSession.getEndDay(), days);
+                
+                DateTime finishedOn = null;
+                String clientTimeZone = null;
+                if (asmtRecord != null) {
+                    clientTimeZone = asmtRecord.getClientTimeZone();
+                    if (clientTimeZone == null) {
+                        clientTimeZone = state.getClientTimeZone();
+                    }
```

Comment:
Client time zone is defined at the top of the document. Only use another time zone if it was recorded in the assessment record, and we want to preserve that information (don't fall back to other time zones...just use Z if it doesn't exist).

Reply:
This now only uses client time zone to adjust the time if it was set on the adherence record in the first place.